### DISCUSSION TITLE
How do resolutions work?

### DISCUSSION THREAD

**Discussion by mel11691** [ClassicOldSong/Apollo#157] [Q&A] [answered]
I was wondering when the virtual display is created, does it max out the possible resolution and refresh rate, and then use windows display settings to bring it down to the moonlight settings?

I ask because I've been missing around with different resolutions a lot on my one client, and it seems like when I try to run a 1440p stream, it gives me a 4k desktop under the hood, but with 1440p selected in the windows manager.

**Comment by ClassicOldSong** [comment, marked as answer]
There's a built-in list of regular resolutions. The resolution will match your client automatically, if it's not in the list it'll create one on the fly.

You don't need to worry about the other resolutions, as Windows will only render at the current selected resolution, so there's no "4k under the hood" happening.

**Reply by mel11691** [reply]
Hey! Thanks for replying. I'm playing Indiana Jones and the Great Circle, which seems to think 4k is native, even though I start moonlight in 1440, or even 1080. I'm trouble shooting issues with the stream and this was one of those things I was checking.

When streaming on moonlight in 1080, I can go into Windows settings and crank the virtual monitor up to 4k with 180hz, though it does say 1080 is recommended in the windows display. So I'm definitely confused by how windows display settings work with virtual monitors.

**Reply by ClassicOldSong** [reply]
Some games do detect resolution wrong, but usually you can go to the game settings and you'll find your current resolution in the list. Sometimes you'll need to change to windowed/borderless mode and set the resolution there and then change back to fullscreen.

For games like Black Myth Wukong tha has hard coded aspect ratio, you might need some mods to get them adapt.

Just downloaded Indiana Jones yesterday, the opening is great!

**Comment by mel11691** [comment]
Oh ya, the voice acting is crazy good.
